File: Python/kyu-7/_0036__Remove_the_minimum/solution.py
def remove_smallest(numbers):
    if numbers:
        min_num = min(numbers)
    else:
        return []
    flag = False
    result = []
    for number in numbers:
        if flag or number != min_num:
            result.append(number)
        else:
            flag = True
    return result

File: JavaScript/kyu-7/_0036__Remove_the_minimum/solution.js
function removeSmallest(numbers) {
  const min = Math.min(...numbers);
  let flag = true;

  return Array.from({ length: numbers.length - 1 }, (_, i) => {
    if (flag) {
      if (numbers[i] !== min) return numbers[i];
      flag = false;
    }
    return numbers[i + 1];
  });
}

console.log(removeSmallest([5, 1, 5, 1]));
